pkg/streamingpromql/aggregations/aggregations_test.go

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package aggregations
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+func TestAggregationGroupNativeHistogramSafety(t *testing.T) {
+	// This test exists to ensure that AggregationGroup implementations correctly remove FloatHistogram instances they retain,
+	// so that retained FloatHistogram instances are not mutated when the HPoint slice they receive is later reused by something else.
```

Comment:
```suggestion
	// so that retained FloatHistogram instances are not mutated when the HPoint slice they received is later reused by something else.
```

Reply:
Addressed in b5ea123.